packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -243,11 +245,18 @@ export class RelevantRunsDataSource {
     if (!isEqual(toCache, this.#cached)) {
       debug('Values changed')
 
+      debug('current cache: %o, new values: %o', this.#cached, toCache)
+
       //TODO is the right thing to invalidate?  Can we just invalidate the runsByCommitShas field?
       const projectSlug = await this.ctx.project.projectId()
 
       await this.ctx.cloud.invalidate('Query', 'cloudProjectBySlug', { slug: projectSlug })
 
+      // If the cache is empty, then we're just starting up. Don't send notifications
```

Comment:
I found this comment a little confusing - it says "if the cache is empty" but the condition is actually the opposite, is the cache NOT empty.

I wonder if we could benefit from a variable here?

```js
const cacheContainsRun = (this.#cached.all.length !== 0 && this.#cached.all[0] && toCache.all[0] && !isEqual(toCache.all[0], this.#cached.all[0])) 

if (cacheContainsRun) {
  ...
}
```

Also can't it just be

```
const cacheContainsRun = (this.#cached.all.length > 0 && !isEqual(toCache.all[0], this.#cached.all[0])) 
```

?

Reply:
The comment is referring to this condition
```js
this.#cached.all.length !== 0
```

As in, if the current cache is empty, then we don't want to send a notification. If we sent one in this case, then users would get notified about whichever run is the latest from the cloud, even if it is like weeks old. 

The comment could use some clarification. Also up for moving these conditions into a variable. I added these
```js
this.#cached.all[0] && toCache.all[0]
```
to satisfy Typescript, because otherwise those could be undefined. If they are, then we don't want to call `maybeSendRunNotification` anyway.

Reply:
```
this.#cached.all.length !== 0 && this.#cached.all[0]
```

I think `this.#cached.all.length !== 0` is not needed, since `this.#cached.all[0]` implies that the length is not 0.